### ISSUE TITLE
DHCPv6 Support?

### ISSUE THREAD

**Issue by Harvie** [misje/dhcpoptinj#21]
Hello, any plans to add DHCPv6 support?

**Comment by misje** [comment]
I was under the impression that IPv6 stateless address autoconfiguration was more common than DHCPv6 setups. I haven't looked into the DHCPv6 protocol and I am not at all familiar with DHCPv6. Can you please provide a scenario where dhcpoptinj makes sense with DHCPv6? It would help researching how DHCPv6 support could be implemented. In general, information about any use of dhcpoptinj is of great interest for me, since I no longer use the utility myself. I need to know how users use it and what they need.

**Comment by Harvie** [comment]
I've been working on deployment of IPv6 in customer IP over Ethernet network in metropolitan ISP. This is how i figured out it should work:

  * Core router uses DHCPv6 with PD (=preffix delegation) to delegate /64 prefix to customer's home wifi router
  * Customer's wifi router then uses stateless setup (RA) to assign addresses from that prefix to individual devices in customers home.

Some middleware like DHCPOPTINJ would then be usefull to inject circuit id and localy add route to routing table. (DHCPv6 only assigns prefix to home router, but core router needs to have route pointing that prefix to that home router)

In other words: stateless autoconfiguration can only provide you with single adress from exactly /64 prefix, but there are many cases, where you need to provide prefix rather than single address, or you don't have whole /64 available to use it for autoconfiguration.
